### PR TITLE
[Fix] Epic moved the Chat system from xmpp to eos connect and its own api

### LINKED ISSUE #775: [EOS Connect] Epic moved the Chat system from xmpp to eos connect and its own api
Tracking issue while i work on the PR.

### PR DESCRIPTION
**Pull request by LeleDerGrasshalmi**
Fixes #775

### CODE REVIEW

File: resources/structs.ts [hunk 1]
```diff
@@ -1434,9 +1452,109 @@ export interface FortniteClientCredentialsAuthData extends AuthData {
...
+  type: 'social.chat.v1.MEMBERS_LEFT';
+}
+
+export interface EOSConnectChatNewMsgMessage extends BaseEOSConnectMessage {
+  payload: {
+    // deployment id
+    namespace: string;
+    conversation: {
+      conversationId: string;
+      type: string; // i.e. 'party
```
Comment (tnfAngel) [inline resources/structs.ts:1516]: smol typo

**Review by tnfAngel — CHANGES_REQUESTED** [review]

File: resources/structs.ts (hunk 1 above)
Comment (LeleDerGrasshalmi) [reply resources/structs.ts:1516]: noone even sees that lol

File: src/stomp/EOSConnect.ts [hunk 2]
```diff
@@ -0,0 +1,226 @@
...
+        heartbeatOutgoing: 30_000,
+        heartbeatIncoming: 0,
+        webSocketFactory: () => new WebSocket(
+          brokerURL,
+          {
+            headers: {
+              Authorization: `Bearer ${this.client.auth.sessions.get(AuthSessionStoreKey.FortniteEOS)!.accessToken}`,
+              'Sec-Websocket-Protocol': 'v10.stomp,v11.stomp,v12.stomp',
+              'Epic-Connect-Protocol': 'stomp',
+              'Epic-Connect-Device-Id': '',
```
Comment (Krowe-moh) [inline src/stomp/EOSConnect.ts:75]: hm?

Comment (LeleDerGrasshalmi) [reply src/stomp/EOSConnect.ts:75]: what=?

**Review by tnfAngel — APPROVED** [review]
I have tested the changes and everything seems to works correctly. For those who upgrade, please note that there is now a length limit of 256 characters when sending a message.

Hoping this great PR gets merged.

**Comment by AntogamerYT** [comment]
image


Works 👍🏻 though it throws an error when the bot is alone in the party:

```js
                    throw new EpicgamesAPIError_1.default(err.response.data, config, err.response.status);
                          ^

EpicgamesAPIError: User [REDACTED] is not member of the conversation
```
(REDACTED being the bot's accountId)

**Comment by LeleDerGrasshalmi** [comment]
hi @AntogamerYT, the latest update should now throw a `PartyChatConversationNotFound` error when there are no other party members.

**Review by ThisNils — CHANGES_REQUESTED** [review]
Looks good!
One small change request 🙂

File: src/exceptions/PartyChatConversationNotFound.ts [hunk 3]
```diff
@@ -0,0 +1,12 @@
+/**
+ * Represents an error thrown because you tried to send a party chat message, but there is no chat yet, because you are the only person in the party
+ */
+class PartyChatConversationNotFound extends Error {
```
[the remark this answers is not in the document]
Comment (LeleDerGrasshalmi) [reply src/exceptions/PartyChatConversationNotFound.ts]: i did change it already in the previous commit, seems like your comment was added on a outdated commit

**Review by ThisNils — APPROVED** [review]

**Pull request merged**